dev-lang/lazarus/lazarus-3.0.ebuild

```diff
@@ -0,0 +1,116 @@
+# Copyright 1999-2023 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+inherit desktop
+
+# See https://wiki.freepascal.org/Lazarus_2.0_fixes_branch for notes and
+# changes in FPCVER. It *does* change between minor versions of lazarus.
+FPCVER="3.2.2"
+
+DESCRIPTION="feature rich visual programming environment emulating Delphi"
+HOMEPAGE="https://www.lazarus-ide.org/"
+SRC_URI="mirror://sourceforge/lazarus/${P}-0.tar.gz"
+
+LICENSE="GPL-2 LGPL-2.1-with-linking-exception"
+SLOT="0/2.2" # Note: Slotting Lazarus needs slotting fpc, see DEPEND.
+KEYWORDS="~amd64 ~x86"
+IUSE="+gui gtk2 gtk qt5 qt6 extras"
+REQUIRED_USE="gui? ( ^^ ( gtk2 gtk qt5 qt6 ) ) extras? ( gui )"
```

Comment:
In a followup (as you've waited long enough), could you make an executive decision as to which framework is preferred? e.g. whatever upstream prefer / what works best?

See https://wiki.gentoo.org/wiki/Project:Qt/Policies https://projects.gentoo.org/qa/policy-guide/use-flags.html#pg0802.

So, for example, if qt6 is best for Lazarus (preferred/default upstream), you would do...
```
IUSE="+gui gtk2 qtk qt5 extras"
REQUIRED_USE="extras? ( gui )"
```

I'm undecided on the REQUIRED_USE for the rest of it and think it should probably go as it doesn't let users easily do "USE=gui qt6" in make.conf without hitting REQUIRED_USE conflicts.

Reply:
you meant 
```
IUSE="+gui gtk2 qtk **qt6** extras"
```

Reply:
No - the idea is that you make one of them controlled by USE=gui.

Reply:
To conclude, if we prefer qt6 we leave qt6 in IUSE

Reply:
@thesamesam see [PR](https://github.com/gentoo/gentoo/pull/37103). Is it the thing you have meant?